pkg/api/v1/types.go

```diff
@@ -1475,6 +1475,28 @@ type PodLogOptions struct {
 	Previous bool `json:"previous,omitempty" description:"return previous terminated container logs; defaults to false"`
 }
 
+// PodAttachOptions is the query options to a Pod's remote attach call
+// TODO: merge w/ PodExecOptions below for stdin, stdout, etc
+type PodAttachOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Stdin if true indicates that stdin is to be redirected for the attach call
+	Stdin bool `json:"stdin,omitempty" description:"redirect the standard input stream of the pod for this call; defaults to false"`
+
+	// Stdout if true indicates that stdout is to be redirected for the attach call
+	Stdout bool `json:"stdout,omitempty" description:"redirect the standard output stream of the pod for this call; defaults to true"`
```

Comment:
Actually, yes. PodExecOptions sets defaults here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/conversion.go#L38

That should not be there. The options should be set by default in each API version, not in the unversioned representation.

Regardless, attach needs to do the same thing.